src/storage/src/file_purger.rs

```diff
@@ -0,0 +1,227 @@
+            &[(1, 1), (2, 2)],
+            &[(Some(1), Some(1)), (Some(2), Some(2))],
+        );
+
+        let iter = memtable.iter(&IterContext::default()).unwrap();
+        let sst_path = "table1";
+        let layer = Arc::new(FsAccessLayer::new(sst_path, os.clone()));
+        let _sst_info = layer
+            .write_sst(sst_file_name, Source::Iter(iter), &WriteOptions {})
+            .await
+            .unwrap();
+
+        (
+            FileHandle::new(
+                FileMeta {
+                    region_id: 0,
+                    file_name: sst_file_name.to_string(),
+                    time_range: None,
+                    level: 0,
+                },
+                layer.clone(),
+                file_purger,
+            ),
+            sst_path.to_string(),
+            layer as _,
+        )
+    }
+
+    #[tokio::test]
+    async fn test_file_purger_handler() {
+        let dir = TempDir::new("file-purge").unwrap();
+        let object_store = ObjectStore::new(
+            Builder::default()
+                .root(dir.path().to_str().unwrap())
+                .build()
+                .unwrap(),
+        );
+        let sst_file_name = "test-file-purge-handler.parquet";
+
+        let noop_file_purger = Arc::new(LocalScheduler::new(
+            SchedulerConfig::default(),
+            NoopFilePurgeHandler,
+        ));
+        let (file, path, layer) =
+            create_sst_file(object_store.clone(), sst_file_name, noop_file_purger).await;
+        let request = FilePurgeRequest {
+            region_id: 0,
+            file_name: file.file_name().to_string(),
+            sst_layer: layer,
+        };
+
+        let handler = FilePurgeHandler;
+        let notify = Arc::new(Notify::new());
+        handler
+            .handle_request(request, Box::new(MockRateLimitToken {}), notify.clone())
+            .await
+            .unwrap();
+
+        notify.notified().await;
+
+        let object = object_store.object(&format!("{}/{}", path, sst_file_name));
+        assert!(!object.is_exist().await.unwrap());
+    }
+
+    #[tokio::test]
+    async fn test_file_purge_loop() {
+        common_telemetry::init_default_ut_logging();
+        let dir = TempDir::new("file-purge").unwrap();
+        let object_store = ObjectStore::new(
+            Builder::default()
+                .root(dir.path().to_str().unwrap())
+                .build()
+                .unwrap(),
+        );
+        let sst_file_name = "test-file-purger.parquet";
+        let scheduler = Arc::new(LocalScheduler::new(
+            SchedulerConfig::default(),
+            FilePurgeHandler,
+        ));
+        let (handle, path, _layer) =
+            create_sst_file(object_store.clone(), sst_file_name, scheduler.clone()).await;
+
+        {
+            // mark file as deleted and drop the handle, we expect the file is deleted.
+            handle.mark_deleted();
+            drop(handle);
+        }
+        scheduler.stop().await.unwrap();
```

Comment:
Maybe we can add a `flush` function for `scheduler` to wait all in-flight tasks finished.

Reply:
> Maybe we can add a `flush` function for `scheduler` to wait all in-flight tasks finished.

I [added an parameter `await_termination: bool` to `Scheduler::stop` method](https://github.com/GreptimeTeam/greptimedb/pull/1030/commits/d5fe3ec8a2577d6cdf9e53097e94a526d3787382#diff-7d1a7794d06be67a5a6cca11de718380af37d859b115c2db7139f996d5087a03R133). If set to true, the scheduler will wait until all queued requests are processed, PTAL.